be/src/formats/parquet/meta_helper.cpp

```diff
@@ -81,28 +145,100 @@ bool ParquetMetaHelper::_is_valid_type(const ParquetField* parquet_field, const
 
     if (parquet_field->type.is_array_type() || parquet_field->type.is_map_type()) {
         for (size_t idx = 0; idx < parquet_field->children.size(); idx++) {
-            if (_is_valid_type(&parquet_field->children[idx], &type_descriptor->children[idx])) {
+            if (_is_valid_type(&parquet_field->children[idx],
+                               physical_schema_field == nullptr ? nullptr : &physical_schema_field->children[idx],
+                               &type_descriptor->children[idx])) {
                 has_valid_child = true;
                 break;
             }
         }
     } else if (parquet_field->type.is_struct_type()) {
-        std::unordered_map<std::string, const TypeDescriptor*> field_name_2_type{};
-        for (size_t idx = 0; idx < type_descriptor->children.size(); idx++) {
-            field_name_2_type.emplace(Utils::format_name(type_descriptor->field_names[idx], _case_sensitive),
-                                      &type_descriptor->children[idx]);
-        }
+        if (physical_schema_field != nullptr) {
+            if (physical_schema_field->__isset.field_id) {
```

Comment:
Do you think if we have a logical_name->physical_schema map, we can deal filed id mapping and name mapping in the same logic?